File: examples/instance.ts
import * as pronote from "../src";
import { credentials } from "./_credentials";

void async function main () {
  const instance = await pronote.instance(credentials.pronoteURL);

  console.log("Name:", instance.name);
  console.log("Server Version:", instance.version, "\n");

  console.log(instance.accounts.length, "available account webspaces");
  for (const account of instance.accounts) {
    console.log("->", account.name, `(${account.path})`);
  }

  if (instance.casURL) {
    console.log(); // Add a line break
    console.log("CAS is activated at", instance.casURL);
    console.log("-> CAS token:", instance.casToken);
  }
}();

File: library/src/commonMain/kotlin/models/Instance.kt
package models

import kotlinx.datetime.LocalDateTime

data class InstanceAccount(
    val name: String,
    val path: String
)

data class Instance(
    val version: List<Int>,
    val name: String,
    val date: LocalDateTime,
    val accounts: List<InstanceAccount>,
    val casURL: String?,
    val casToken: String?
)
